tinytodo-go/internal/app/server/httphandlersapi.go

```diff
@@ -0,0 +1,1493 @@
+//
+// Example response:
+//
+//	[
+//		{'uid': 'List::"0"', 'owner': 'User::"andrew"', 'name': 'Cedar blog post', 'tasks': [], 'readers': 'Team::"1"', 'editors': 'Team::"2"'},
+//		{'uid': 'List::"1"', 'owner': 'User::"aaron"', 'name': 'my task list', 'tasks': [], 'readers': 'Team::"3"', 'editors': 'Team::"4"'}
+//	]
+func (s *Server) handleGetAPIListsGet(w http.ResponseWriter, r *http.Request) {
+
+	userUID_, err := entitystore.ParseEntityUID(r.URL.Query().Get("uid"))
+	if err != nil {
+		s.logger.InfoContext(
+			r.Context(),
+			"failed to process uid",
+			slog.Any("error", err),
+			slog.String("uid", r.URL.Query().Get("uid")),
+		)
+		s.writeGenericResponse(
+			r.Context(),
+			w,
+			&GenericResponse{
+				Error: "Failed to process uid",
+			},
+			http.StatusBadRequest,
+		)
+		return
+	}
+
+	userUID := entitystore.UserUID{EntityUID: userUID_}
+
+	s.logger.InfoContext(
+		r.Context(),
+		"processing get lists request",
+		slog.Any("userUID", userUID),
+	)
+
+	decision, diagnostic, err := s.isAuthorized(
+		r.Context(),
+		userUID.EntityUID,
+		action.GetLists,
+		ApplicationEntityUID.EUID,
+	)
+	if err != nil {
+		s.logger.ErrorContext(
+			r.Context(),
+			"failed to check authorization",
+			slog.Any("principal", userUID),
+			slog.Any("action", action.GetLists),
+			slog.Any("resource", ApplicationEntityUID),
+			slog.Any("error", err),
+		)
+		s.ServerError(w, err)
+		return
+	}
+
+	if !decision {
+		s.logger.InfoContext(
+			r.Context(),
+			"not allowed",
+			slog.Any("principal", userUID),
+			slog.Any("action", action.GetLists),
+			slog.Any("resource", ApplicationEntityUID),
+			slog.Any("diagnostic", diagnostic),
+		)
+		s.writeGenericResponse(
+			r.Context(),
+			w,
+			&GenericResponse{
+				Error: "Authorization Denied", // this particular error message is required by tinytodo.py
+			},
+			http.StatusOK, // tinytodo.py expects 200 for authorization denied
+		)
+		return
+	}
+
+	// client expects allowedLists to be an empty list (vs nil), don't convert to var declaration
+	allowedLists := []*entitystore.List{}
+
+	for listEUID, list := range s.es.Lists {
+		listAllowed, _, err := s.isAuthorized(
+			r.Context(),
+			userUID.EntityUID,
+			action.GetList,
+			listEUID.EntityUID,
+		)
+		if err != nil {
+			s.logger.ErrorContext(
+				r.Context(),
+				"failed to check authorization",
+				slog.Any("userUID", userUID),
+				slog.Any("listEUID", listEUID),
+				slog.Any("error", err),
+			)
+			s.ServerError(w, err)
+			return
+		}
+		if !listAllowed {
+			continue
+		}
+		allowedLists = append(allowedLists, list)
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(allowedLists)
+}
+
+type apiListCreateRequest struct {
+	UID  string `json:"uid"`
+	Name string `json:"name"`
+}
+
+// handlePostAPIListCreate handles POST requests to "/api/list/create".
+//
+// Example response:
+//
+//	List::"0"
+func (s *Server) handlePostAPIListCreate(w http.ResponseWriter, r *http.Request) {
```

Comment:
Looks like there's ~50 lines of calling Cedar at each auth endpoint. It might be a better demo if we can abstract that somehow? Also helps with maintenance and ensuring you don't forget to call Cedar